# Trabajo Práctico 1 - Matrices de Insumo Producto

## Enunciado

### Consigna 1 - 

#### a) $(I - A)$ es inversible:

Si partimos de la ecuacion [1] : 

$p = A*p + d \Rightarrow p - A*p = d \Rightarrow (I - A)p = d$

Entonces, si queremos despejar $p$ necesitamos que $(I - A)$ sea inversible.

Por lo que, resolveremos la siguiente multiplicacion para hallar $p$ :

$(I - A)^{-1}d = p$ ,donde $A$ y $d$ son datos.




#### b) $(I - A)$ es inversible y $d = 0$:

Si solo analizamos la ecuacion [2]:

tendriamos a $p = (I-A)^{-1}0 \Rightarrow p = 0$.

#### c) $(I - A)$ no es inversible
Como notamos en el enciso a) : cuando $(I-A)p = d$ 

necesitamos que $(I-A)$ sea inversible para poder expresar la ecuacion [2].

Pero, igualmente la solucion a nuestro problema sigue vigente a pesar de que $(I-A)$ no tenga inversa.

Por lo que, calcular con eliminacion Gaussiana la solucion de la ecuacion $(I-A)p = d$, es una posibilidad.


#### d) $(I - A)$ no es inversible y $d = 0$
En esta oportunidad, solo tendriamos la alternativa de calcular la solucion de la ecuacion:

$(I - A)p = d$

siendo que $d=0$ , estariamos diciendo que $p \in Nu(I-A)$

---

### Consigna 2 -  

In [147]:
import funciones as f

---

### Consigna 3 - 

Construimos la matriz A :

In [95]:
A = [[0.3,0.0,0.1],[0.5,1.0,0.2],[0.1,0.15,0.1]]

Contruimos la matriz d:

In [97]:
d = [100,100,300]

Primeramente, usamos la funcion calcularLU() para obtener L y U

In [117]:
I = np.ones(3)
L, U, P = f.calcularLU(A - I)
print("Matriz L:\n", L)
print("Matriz U:\n", U)
print("Matriz P:\n", P)

Matriz L:
 [[ 1.  0.  0.]
 [-0.  1.  0.]
 [ 0. -0.  1.]]
Matriz U:
 [[-0.9        -0.85       -0.9       ]
 [ 0.         -0.33888889 -0.2       ]
 [-0.5         0.         -0.8       ]]
Matriz P:
 [[0. 0. 1.]
 [1. 0. 0.]
 [0. 1. 0.]]


Una vez, con las matrices L y U calculamos la inversa de A

In [119]:
Inv = f.inversaLU(L, U, P)
print("Inversa de A:\n", Inv)

Inversa de A:
 [[-1.53604029  3.85269122  0.76487252]
 [-0.56657224 -1.52974504  1.01983003]
 [ 0.96002518 -2.40793201 -1.72804533]]


Por ultimo, resolvemos el sistema

In [125]:
p = np.dot(Inv, d)
print("Vector p:\n", p)

Vector p:
 [ 461.12684923   96.31728045 -663.20428077]


#### ¿Hay una justificación económica de los valores encontrados para $p$?
Los valores del vector $p$, que son $[461.13,96.32,−663.20]$, representan las cantidades que deben producir los sectores para satisfacer tanto la demanda interna como la externa.

Entonces, los valores reflejan la necesidad de ajustes en la producción de los sectores para satisfacer las demandas de manera eficiente y podrían indicar que el modelo necesita revisión para evitar resultados no realistas.


#### ¿Cuál es la característica de la matriz que lleva a esta solución?

la característica principal de la matriz es su posible mala condición, que contribuye a la obtención de resultados inestables o no realistas, como el valor negativo en la producción.

---

### Consigna 4 -

El cambio en la demanda externa es representado por $\Delta d_3$, dado que solo afecta para el producto 3.

Lo podemos expresar como $\Delta d_3 = d' - d \Rightarrow d' = d + \Delta d_3$ 

El cambio de la produccion total se puede expresar como : $\Delta p = p'- p$

donde $p' = Ld'$ y $p = Ld$ , siendo $L$ la matriz de Leonief.

Entonces, $\Delta p = Ld' - Ld \Rightarrow \Delta p = (Ld + L\Delta d_3) - Ld  \Rightarrow \therefore \Delta p = L\Delta d_3$

Este resultado significa que el cambio en la producción de cada sector está determinado por cómo está relacionado ese sector con el producto 3 (representado por la tercera columna de la matriz $L$)